nuxeo-mongodb-directory/src/main/java/org/nuxeo/directory/mongodb/MongoDBSession.java

```diff
@@ -222,9 +327,42 @@ public DocumentModelList query(Map<String, Serializable> filter, Set<String> ful
             results.limit(limit);
         }
         for (Document resultDoc : results) {
+
             // Cast object to document model
             Map<String, Object> fieldMap = MongoDBSerializationHelper.bsonToFieldMap(resultDoc);
-            entries.add(fieldMapToDocumentModel(fieldMap));
+            DocumentModel doc = fieldMapToDocumentModel(fieldMap);
+
+            if (fetchReferences) {
+                Map<String, List<String>> targetIdsMap = new HashMap<>();
+                for (Reference reference : directory.getReferences()) {
+                    List<String> targetIds;
+                    if (reference instanceof MongoDBReference) {
+                        MongoDBReference mongoReference = (MongoDBReference) reference;
+                        targetIds = mongoReference.getTargetIdsForSource(doc.getId(), this);
+                    }
+                    else {
+                        targetIds = reference.getTargetIdsForSource(doc.getId());
+                    }
+                    targetIds = new ArrayList<>(targetIds);
+                    Collections.sort(targetIds);
```

Comment:
Why do you need to sort them ? If `targetIdsMap` already contains the field, the resulting order in the map won't be correct regarding this line.